src/GraphQL/Execution/ExecutionStrategy.cs

```diff
@@ -84,28 +84,25 @@ public static void SetSubFieldNodes(ExecutionContext context, ObjectExecutionNod
         /// <summary>
         /// Creates specified child execution nodes of an object execution node.
         /// </summary>
-        public static void SetSubFieldNodes(ExecutionContext context, ObjectExecutionNode parent, Fields fields)
+        private static void SetSubFieldNodes(ExecutionContext context, ObjectExecutionNode parent, Fields fields)
         {
             var parentType = parent.GetObjectGraphType(context.Schema);
 
-            var subFields = new Dictionary<string, ExecutionNode>(fields.Count);
+            var subFields = new ExecutionNode[fields.Count];
 
+            int i = 0;
             foreach (var kvp in fields)
             {
-                var name = kvp.Key;
                 var field = kvp.Value;
 
                 var fieldDefinition = ExecutionHelper.GetFieldDefinition(context.Schema, parentType, field);
 
                 if (fieldDefinition == null)
-                    continue;
```

Comment:
@Shane32 See `UnionInterfaceTests.cs`. I changed two tests. Previously, these tests succeeded due to that `continue`. I think it's wrong. Note that both tests pass empty validation list to bypass the validation stage. If you change the parameter to `null`, then an error occurs at the validation stage. And this is a correct error, because the request is incorrect. I believe the error should be thrown at runtime even for bypassed validation. @joemcbride Could you please comment?

Reply:
This change makes sense to me.